### PR TITLE
fixed focus issue on form validation

### PR DESCRIPTION
**Pull request by sunilit42**
Fixed Issues (if relevant) : https://github.com/magento/magento2/issues/24045

### CODE REVIEW

**Comment by sunilit42** [comment]
@orlangur  can you please tell me why static tests fail?

**Review by VladimirZaets — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @VladimirZaets, thank you for the review.
[ENGCOM-5578](https://jira.corp.magento.com/browse/ENGCOM-5578) has been created to process this Pull Request
:eight_spoked_asterisk: @VladimirZaets, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by engcom-Alfa** [comment]
Hi @sunilit42 !

During testing we faced some issues.

 **_First problem:_**  After pressing the "Save" button again, the focus on the attribute disappears

**Actual Result:**

Peek 2019-08-12 10-54

**_Second problem:_** Scrolling time is too slow
Setting the scroll time will affect to the MFTF tests. We think that it is not correct behavior

**Actual Result:**
screen3

**Expected Result:**
screen2

@sunilit42  Could you take a look?

Thanks!

**Comment by sunilit42** [comment]
@engcom-Alfa fix the issue by adding a timeout for form element focus

we need to add time out because  fieldset.js calling after that and that js open that tab so

@VladimirZaets  please check and approve it :)

**Comment by sunilit42** [comment]
Any update here?

**Comment by sunilit42** [comment]
Looking forward reply @engcom-Alfa

**Comment by engcom-Alfa** [comment]
Hi @sunilit42 .

Sorry, but we are waiting for approval from reviewer @VladimirZaets

Thanks!

**Comment by VladimirZaets** [comment]
@magento give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @VladimirZaets. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @VladimirZaets, here is your [Magento instance](https://i-24090-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-24090-2-3-develop.instances.magento-community.engineering/admin](https://i-24090-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

File: app/code/Magento/Ui/view/base/web/js/form/form.js
Comment (VladimirZaets) [inline app/code/Magento/Ui/view/base/web/js/form/form.js:281]: Can you please explain why you use setTimeout here and why we need to wait for the magic 2 milliseconds? 

Moreover, you fix doesn't work on my local instance.

**Review by VladimirZaets — CHANGES_REQUESTED** [review]
Hi @sunilit42. Thanks for collaboration. 
Please look at my code review comments.

File: app/code/Magento/Ui/view/base/web/js/form/form.js
Comment (sunilit42) [reply app/code/Magento/Ui/view/base/web/js/form/form.js:281]: @VladimirZaets 
https://github.com/magento/magento2/blob/2.3-develop/app/code/Magento/Ui/view/base/web/js/form/components/fieldset.js#L167

fieldset.js open that tab then focus on element will work

Hope it will make sense

can you please confirm my js changes loaded?

**Comment by sunilit42** [comment]
@magento give me test instance

**Comment by magento-engcom-team** [comment]
Hi @sunilit42. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @sunilit42, here is your [new Magento instance](https://pr-24090.instances.magento-community.engineering).
Admin access: [https://pr-24090.instances.magento-community.engineering/admin](https://pr-24090.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`

**Comment by sunilit42** [comment]
@magento give me test instance

**Comment by magento-engcom-team** [comment]
Hi @sunilit42. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @sunilit42, here is your [new Magento instance](https://pr-24090.instances.magento-community.engineering).
Admin access: [https://pr-24090.instances.magento-community.engineering/admin](https://pr-24090.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`

**Comment by sunilit42** [comment]
@magento give me test instance

**Comment by magento-engcom-team** [comment]
Hi @sunilit42. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @sunilit42, here is your [new Magento instance](https://pr-24090.instances.magento-community.engineering).
Admin access: [https://pr-24090.instances.magento-community.engineering/admin](https://pr-24090.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`

**Comment by sunilit42** [comment]
@magento give me test instance

**Comment by magento-engcom-team** [comment]
Hi @sunilit42. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @sunilit42, here is your [new Magento instance](https://pr-24090.instances.magento-community.engineering).
Admin access: [https://pr-24090.instances.magento-community.engineering/admin](https://pr-24090.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`

**Comment by sunilit42** [comment]
@VladimirZaets 

please check now it is working now

File: app/code/Magento/Ui/view/base/web/js/form/form.js
Comment (sunilit42) [reply app/code/Magento/Ui/view/base/web/js/form/form.js:281]: @VladimirZaets  please check it now

Comment (VladimirZaets) [reply app/code/Magento/Ui/view/base/web/js/form/form.js:281]: @sunilit42 So maybe we should fix it globally, because, looks like have the same ( wrong ) behavior in all cases where we have dropdown and validation

**Comment by sunilit42** [comment]
@magento give me test instance

**Comment by magento-engcom-team** [comment]
Hi @sunilit42. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @sunilit42, here is your [new Magento instance](https://pr-24090.instances.magento-community.engineering).
Admin access: [https://pr-24090.instances.magento-community.engineering/admin](https://pr-24090.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`

File: app/code/Magento/Ui/view/base/web/js/form/form.js
Comment (sunilit42) [reply app/code/Magento/Ui/view/base/web/js/form/form.js:281]: @VladimirZaets  then i can add if more than one tab then use mode code otherwise as default one

Comment (VladimirZaets) [reply app/code/Magento/Ui/view/base/web/js/form/form.js:281]: @sunilit42, it's not a valid case in general, we should do an action when some action happened, we should investigate how we can manage this situation. In your case if someone will change the configuration of dropdown ( time of opening to 2 sec for example) this bug again will be reproducible. 

Otherwise, if we need to move call this function to the end of the call stack we need to make setTimeout 0.

Comment (sunilit42) [reply app/code/Magento/Ui/view/base/web/js/form/form.js:281]: There are a lot of places Magento itself using timeout

Comment (VladimirZaets) [reply app/code/Magento/Ui/view/base/web/js/form/form.js:281]: @sunilit42 a lot of legacy code in Magento that currently, we don't recommend to use.

**Comment by VladimirZaets** [comment]
Hi @sunilit42. It's incorrect fix, you try 3 times to set different time to timeout to fix this issue. In another case with other conditions, we will have different times for a timeout. It's not stable fix, we cannot be sure that you fix will work in all cases

**Comment by krzksz** [comment]
Closing due to inactivity and proposed solution not being accepted. Feel free to open if you wish to continue working on this pull request.

**Pull request closed without merging**
